web/client/epics/backgroundselector.js

```diff
@@ -36,14 +37,16 @@ import { getCustomTileGridProperties, getLayerOptions } from '../utils/WMSUtils'
 import { getLayerTileMatrixSetsInfo } from '../api/WMTS';
 import { generateGeoServerWMTSUrl } from '../utils/WMTSUtils';
 
-const accessMetadataExplorer = (action$) =>
+const accessMetadataExplorer = (action$, store) =>
     action$.ofType(ADD_BACKGROUND)
         .switchMap(() => Rx.Observable.of(
             setControlProperty('metadataexplorer', 'enabled', true),
             allowBackgroundsDeletion(false),
+            addBackupBackground(store.getState().catalog.selectedService),
```

Comment:
Use selectors when available  in this case `selectedServiceSelector` in `selectors/catalog`.
This helps future refactors (e.g. internal restructuration of state) without having to find around the code all the `x.y.z` strings.